### PR TITLE
Add DVC team & DVC team modal to team management page

### PR DESCRIPTION
**Pull request by ajspotts**
Part 3 of stack
Part 1 #14981 
Part 2 #14982 

### Description
This PR adds the frontend additions for the new DVC team including the changes to the team management page and the DVC modal.

### Acceptance Criteria
- [x] Can successfully add a new DVC team and add/remove members from the team

### Testing Plan
1. Sign in as TEAM_ADMIN
2. Go to http://localhost:3000/team_management
3. Make sure the DVC teams header and +Add DVC Team button are present
4. Click on Add DVC Team and make sure modal appears
5. Select a user to create a DVC team for (remember the css_id you choose) and hit Submit
6. Ensure the Success banner appears and a new team is listed under DVC teams
7. Go to the Org admin page link and make sure you can add & remove team members

#### BEFORE
Screen Shot 2020-08-14 at 3 38 47 PM

#### AFTER
Screen Shot 2020-08-14 at 3 38 29 PM

#### AFTER
Screen Shot 2020-08-19 at 6 29 21 PM

### CODE REVIEW

File: client/app/components/SearchableDropdown.jsx [hunk 1]
```diff
@@ -157,7 +157,7 @@ export class SearchableDropdown extends React.Component {
     // because if the selector filters the options to be [], it will show the "no results found"
     // message. We can get around this by unsetting `noResultsText`.
     const handleNoOptions = () =>
-      noResultsText ?? creatable ? null : NO_RESULTS_TEXT;
+      noResultsText ?? (creatable ? null : NO_RESULTS_TEXT);
```
Comment (ajspotts) [inline client/app/components/SearchableDropdown.jsx:161]: Fixes order of operations issue which was preventing the statement from executing properly

File: client/app/components/SearchableDropdown.jsx [hunk 2]
```diff
@@ -157,7 +157,7 @@ export class SearchableDropdown extends React.Component {
...
```
Comment (ajspotts) [inline client/app/components/SearchableDropdown.jsx:160]: Fixes order of operations issue which was preventing the statement from executing properly

File: client/app/components/SearchableDropdown.jsx [hunk 3]
```diff
@@ -237,6 +237,10 @@ SearchableDropdown.propTypes = {
   loading: PropTypes.bool,
   multi: PropTypes.bool,
   name: PropTypes.string.isRequired,
+
+  /**
+ * react-select will by default set noResultsText to say "No options" unless the prop is explicitly defined
+ */
```
Comment (ajspotts) [inline client/app/components/SearchableDropdown.jsx:243]: The message "No options" in the SearchableDropdown component comes from our react-select dependency and needs to be manually overwritten

File: client/app/components/SearchableDropdown.stories.js [hunk 4]
```diff
@@ -135,3 +135,10 @@ export const async = () => {
     <SearchableDropdown name="async" label="Select an Option" async={asyncFn} />
   );
 };
+
+export const NoResultsText = Template.bind({});
+NoResultsText.args = {
+  name: 'noResultsText',
+  options: [],
+  noResults: 'Your results are in another castle'
+};
```
Comment (ajspotts) [inline client/app/components/SearchableDropdown.stories.js:144]: Add Storybook story for the SearchableDropdown `noResultsText` prop

File: client/app/components/SearchableDropdown.jsx [hunk 5]
```diff
@@ -237,6 +237,10 @@ SearchableDropdown.propTypes = {
...
   noResultsText: PropTypes.string,
```
Comment (ajspotts) [inline client/app/components/SearchableDropdown.jsx:244]: The message "No options" in the SearchableDropdown component comes from our react-select dependency and needs to be manually overwritten in the noResultsText prop

File: client/app/queue/OrganizationUsers.jsx [hunk 6]
```diff
@@ -265,6 +267,7 @@ export default class OrganizationUsers extends React.PureComponent {
             `${COPY.USER_MANAGEMENT_ADD_USER_LOADING_MESSAGE} ${this.formatName(this.state.addingUser)}` :
             COPY.USER_MANAGEMENT_ADD_USER_TO_ORG_DROPDOWN_TEXT
         }
+        noResultsText={COPY.TEAM_MANAGEMENT_DROPDOWN_LABEL}
```
Comment (ajspotts) [inline client/app/queue/OrganizationUsers.jsx:270]: Here's where we override that "No options" text with something a little cleaner

File: client/app/queue/TeamManagement.jsx [hunk 7]
```diff
@@ -21,11 +21,17 @@ const tableStyling = css({
...
+const buttonStyling = css({
+  marginLeft: '1rem'
+});
+
 class TeamManagement extends React.PureComponent {
   loadingPromise = () => ApiUtil.get('/team_management').then((resp) => this.props.onReceiveTeamList(resp.body));
 
   addJudgeTeam = () => this.props.history.push('/team_management/add_judge_team');
 
+  addDvcTeam = () => this.props.history.push('/team_management/add_dvc_team');
```
Comment (ajspotts) [inline client/app/queue/TeamManagement.jsx:33]: redirect to our new DVC modal

File: client/COPY.json [hunk 8]
```diff
@@ -463,15 +463,24 @@
   "TEAM_MANAGEMENT_URL_COLUMN_HEADING": "URL",
   "TEAM_MANAGEMENT_PARTICIPANT_ID_COLUMN_HEADING": "BGS Participant ID",
   "TEAM_MANAGEMENT_UPDATE_ROW_BUTTON": "Update",
+  "TEAM_MANAGEMENT_ADD_JUDGE_LABEL": "Judge Teams",
```
Comment (lomky) [inline client/COPY.json:466]: Great call cleaning up all these strings!

File: client/app/queue/OrganizationUsers.jsx [hunk 9]
```diff
@@ -42,6 +42,8 @@ const listStyle = css({
   listStyle: 'none'
 });
 
+// const dropdownLabel = 'Please enter a valid name or CSS ID to search for a user';
+
```
[the remark this answers is not in the document]
Comment (ajspotts) [reply client/app/queue/OrganizationUsers.jsx:46]: Ah! Sorry that was a remnant of me testing the SearchableDropdown component prior to extraction to `COPY.json`. Removing!

**Pull request merged**